src/parser/view_transform.c

```diff
@@ -1542,26 +1542,57 @@ mq_update_order_by (PARSER_CONTEXT * parser, PT_NODE * statement, PT_NODE * quer
       /* if attr is not found in output list, append a hidden column at the end of the output list. */
       if (node == NULL)
 	{
-	  result = parser_copy_tree (parser, attr);
-	  if (result == NULL)
+	  if (pt_is_distinct (statement))
 	    {
-	      PT_INTERNAL_ERROR (parser, "parser_copy_tree");
-	      return NULL;
+	      /* remove unnecessary order */
+	      if (prev_order == NULL)
+		{
+		  statement->info.query.order_by = order->next;
+		}
+	      else
+		{
+		  prev_order->next = order->next;
+		}
+	      /* free order */
+	      save_next = order->next;
+	      order->next = NULL;
+	      parser_append_node (order, free_node);
```

Comment:
I think that the "free_node" never has any node entry but initial NULL setting.

Reply:
Actually, the
return value of parser_append_node() is unnecessary. the pointer of free_node can be changed through parser_append_node().

Reply:
If the "free_node" has no value pointer then it's not necessary to be free it as the last statement.

Reply:
I made a mistake. It's fixed. thank you.